arm9/modules/59/src/mod59_021D74E0_src.c

```diff
@@ -277,3 +280,44 @@ THUMB_FUNC void MOD59_SetupBg(MOD59_OverlayData *data)
     MOD59_021D8058(data);
     data->Unk78 = 0;
 }
+
+THUMB_FUNC void MOD59_DestroyBg(MOD59_OverlayData *data)
+{
+    ToggleBgLayer(GF_BG_LYR_MAIN_0, GX_LAYER_TOGGLE_OFF);
+    ToggleBgLayer(GF_BG_LYR_MAIN_1, GX_LAYER_TOGGLE_OFF);
+    ToggleBgLayer(GF_BG_LYR_MAIN_2, GX_LAYER_TOGGLE_OFF);
+    ToggleBgLayer(GF_BG_LYR_MAIN_3, GX_LAYER_TOGGLE_OFF);
+
+    ToggleBgLayer(GF_BG_LYR_SUB_0, GX_LAYER_TOGGLE_OFF);
+    ToggleBgLayer(GF_BG_LYR_SUB_1, GX_LAYER_TOGGLE_OFF);
+    ToggleBgLayer(GF_BG_LYR_SUB_2, GX_LAYER_TOGGLE_OFF);
+    ToggleBgLayer(GF_BG_LYR_SUB_3, GX_LAYER_TOGGLE_OFF);
+
+    FreeBgTilemapBuffer(data->bgConfig, GF_BG_LYR_MAIN_0);
+    FreeBgTilemapBuffer(data->bgConfig, GF_BG_LYR_MAIN_1);
+    FreeBgTilemapBuffer(data->bgConfig, GF_BG_LYR_MAIN_2);
+    FreeBgTilemapBuffer(data->bgConfig, GF_BG_LYR_MAIN_3);
+
+    FreeBgTilemapBuffer(data->bgConfig, GF_BG_LYR_SUB_0);
+    FreeBgTilemapBuffer(data->bgConfig, GF_BG_LYR_SUB_1);
+    FreeBgTilemapBuffer(data->bgConfig, GF_BG_LYR_SUB_2);
+    FreeBgTilemapBuffer(data->bgConfig, GF_BG_LYR_SUB_3);
+
+    FreeToHeap(data->bgConfig);
+}
+
+THUMB_FUNC void MOD59_SetupMsg(MOD59_OverlayData *data)
+{
+    data->msgData = NewMsgDataFromNarc(1, NARC_MSGDATA_MSG, 0x155, data->heap_id);
```

Comment:
```suggestion
    data->msgData = NewMsgDataFromNarc(1, NARC_MSGDATA_MSG, NARC_msg_narc_0341_bin, data->heap_id);
```
you'll need to `#include "msgdata/msg.naix"`